### ISSUE TITLE
Crashing on interactively defined functions (jupyter notebook): AttributeError: Can't get attribute on <module

### ISSUE THREAD

**Issue by DreamFlasher** [joblib/joblib#659]
```
from joblib import Parallel, delayed
def f(x): return x**2
Parallel(n_jobs=3)(delayed(f)(i) for i in range(10))
```

yields:
```
  File "...\lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'f' on <module '__main__' (built-in)>
```

Environment:

joblib: 0.11
Python: 3.6.1 Anaconda
Jupyter Notebook
Windows 10

**Comment by lesteve** [comment]
We are using cloudpickle in master (rather than pickle) so this should be fixed in master, can you double-check?

**Comment by altosaar** [comment]
+1, also running into this.

**Comment by ogrisel** [comment]
Have you tried with joblib 0.12? This should be fixed. If so let's close this.

**Comment by DreamFlasher** [comment]
I can confirm that this is fixed in 0.12. Thank you!

**Comment by ogrisel** [comment]
Thanks for the confirmation.

**Comment by albertcthomas** [comment]
I ran into the same issue and I can also confirm that this is fixed in 0.12
